### PR TITLE
[proc] add serialization for process stats that system-probe collects

### PR DESCRIPTION
**Pull request by shang-wang**
### What does this PR do?

This PR is the continuation of this PR: https://github.com/DataDog/datadog-agent/pull/7314, it mainly adds the serialization methods to prepare for process-agent to read data from the endpoint.

@DataDog/processes 

### Motivation

We want customers to have those metrics without granting extra permission to process-agent.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

File: pkg/process/encoding/json.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	}
+	return stats, nil
+}
+
+func (j jsonSerializer) ContentType() string {
+	return ContentTypeJSON
+}
+
+var _ Marshaler = jsonSerializer{}
+var _ Unmarshaler = jsonSerializer{}
```
Comment (robertjli) [inline pkg/process/encoding/json.go:54]: Out of curiosity, what do these lines do?

**Review by robertjli — APPROVED** [review]

File: pkg/process/encoding/json.go (hunk 1 above)
Comment (shang-wang) [reply pkg/process/encoding/json.go:54]: https://golang.org/doc/faq#guarantee_satisfies_interface This ensures that the struct implements the interface when you compile it.

File: go.mod [hunk 2]
```diff
@@ -40,7 +40,7 @@ require (
 	code.cloudfoundry.org/rep v0.0.0-20200325195957-1404b978e31e // indirect
 	code.cloudfoundry.org/rfc5424 v0.0.0-20180905210152-236a6d29298a // indirect
 	code.cloudfoundry.org/tlsconfig v0.0.0-20200131000646-bbe0f8da39b3 // indirect
-	github.com/DataDog/agent-payload v4.55.0+incompatible
+	github.com/DataDog/agent-payload v4.55.1-0.20210129231257-367e91ca1f0f+incompatible
```
Comment (mbotarro) [inline go.mod:43]: Are we going to update it once the agent-payload PR is merged and a new release is created?

File: cmd/system-probe/modules/process_tracer.go [hunk 3]
```diff
@@ -87,3 +90,16 @@ func logProcTracerRequests(count uint64, statsCount int, start time.Time) {
 		log.Debugf(msg, args...)
 	}
 }
+
+func writeStats(w http.ResponseWriter, marshaler encoding.Marshaler, stats map[int32]*procutil.StatsWithPerm) {
+	buf, err := marshaler.Marshal(stats)
+	if err != nil {
+		log.Errorf("unable to marshall stats with type %s: %s", marshaler.ContentType(), err)
```
Comment (mbotarro) [inline cmd/system-probe/modules/process_tracer.go:97]: nit: r/marshall/marshal

File: pkg/process/encoding/encoding_test.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+		assert.Equal(t, "application/json", marshaler.ContentType())
+
+		var empty map[int32]*procutil.StatsWithPerm
+		blob, err := marshaler.Marshal(empty)
+		require.NoError(t, err)
+
+		unmarshaler := GetUnmarshaler("application/json")
+		result, err := unmarshaler.Unmarshal(blob)
+		require.NoError(t, err)
+		assert.EqualValues(t, &model.ProcStatsWithPermByPID{StatsByPID: map[int32]*model.ProcStatsWithPerm{}}, result)
```
Comment (mbotarro) [inline pkg/process/encoding/encoding_test.go:123]: Just as a curiosity, why do we need to initialize an empty `StatsByPID` with the json marshaler but not with the protobuf one?

**Review by mbotarro — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (shang-wang) [reply go.mod:43]: Yes! 👍

File: pkg/process/encoding/encoding_test.go (hunk 4 above)
Comment (shang-wang) [reply pkg/process/encoding/encoding_test.go:123]: I'm not sure, it probably has something to do with the serialization code difference for each. The json one does this but not protobuf.

**Review by heyronhay — APPROVED** [review]

**Pull request merged**